docs/runtimes/image-build.md

```diff
@@ -0,0 +1,192 @@
+resulting image name, as specified in the `image` parameter.
+
+```python
+project.build_function(
+   "trainer",
+   base_image="myrepo/my_base_image:latest",
+)
+```
+
+### Running commands
+To run arbitrary commands during the image build, pass them in the `commands` parameter of 
+{py:func}`~mlrun.projects.build_function()`. For example:
+
+```python
+github_repo = "myusername/myrepo.git@mybranch"
+
+project.build_function(
+   "trainer",
+   base_image="myrepo/base_image:latest",
+   commands= [
+        "pip install git+https://github.com/" + github_repo,
+        "mkdir -p /some/path && chmod 0777 /some/path",    
+   ]
+)
+```
+
+These commands will be added as `RUN` operations to the dockerfile generating the image.
+
+### MLRun package deployment
+The `with_mlrun` and `mlrun_version_specifier` parameters allows control over the inclusion of the MLRun package in the
+build process. Depending on the base-image used for the build, the MLRun package may already be available in which 
+case use `with_mlrun=False`. If not specified, MLRun will attempt to detect this situation - if the image used is one 
+of the default MLRun images released with MLRun, `with_mlrun` will automatically be set to `False`.
+If the code executing requires a different version of MLRun than the one used to deploy the function, 
+set the `mlrun_version_specifier` to point at the specific version needed. This will use the published MLRun images
+of the specified version instead. 
+For example:
+
+```python
+project.build_function(
+   "trainer",
+   with_mlrun=True,
+   mlrun_version_specifier="1.0.0"
+)
+```
+
+### Working with code repository
+As the code matures and evolves, the code will usually be stored in a git code repository.
+When the MLRun project is associated with a git repo (see {ref}`create-projects` for details), functions can be added
+by calling {py:func}`~mlrun.projects.MlrunProject.set_function()` and setting `with_repo=True`. This indicates that the 
+code of the function should be retrieved from the project code repository.
+
+In this case, the entire code repository will be retrieved from git as part of the image-building process, and cloned
+into the built image. This is recommended when the function relies on code spread across multiple files and also is 
+usually preferred for production code, since it means that the code of the function is stable, and further modifications 
+to the code will not cause instability in deployed images.
+
+During the development phase it may be desired to retrieve the code in runtime, rather than re-build the function
+image every time the code changes. To enable this, use {py:func}`~mlrun.projects.MlrunProject.set_source()` which
+gets a path to the source (can be a git repository or a tar or zip file) and set `pull_at_runtime=True`.
+
+### Using a private Docker registry
+By default, images will be pushed to the registry configured during MLRun deployment, using the registry credentials 
```

Comment:
By default, images are pushed to the registry configured during MLRun deployment, using the configured registry credentials .